zubbi/scraper/scraper.py

```diff
@@ -146,39 +160,76 @@ def get_file_info(self, path):
-            for role_name, remote_file in roles.items():
+            # Ansible requires the role to be defined in a certain directory structure.
+            # Thus, we are only interested in directories. Files can be omitted
+            dir_contents = [
+                item
+                for item in self.repo.directory_contents(ROLES_DIRECTORY).values()
+                if item.type == "dir"
+            ]
+            while dir_contents:
                 try:
-                    if remote_file.type != "dir":
-                        # Ansible requires the role to be defined in a
-                        # certain directory structure. Thus, we can
-                        # ignore it in case it's not a directory.
-                        continue
-                    last_changed = self.repo.last_changed(remote_file.path)
-                    existing_files = self.repo.directory_contents(remote_file.path)
-                    # Skip empty directories or files
-                    if not existing_files:
-                        continue
-                    readme_file = self.find_matching_file(README_FILES, existing_files)
-                    changelog_file = self.find_matching_file(
-                        CHANGELOG_FILES, existing_files
-                    )
-                    role_files[role_name] = {
-                        "last_changed": last_changed,
-                        "readme_file": readme_file,
-                        "changelog_file": changelog_file,
-                    }
+                    dir = dir_contents.pop(0)
+
+                    # When directory is one of role mandatory directories we assume that
+                    # its parent is the role directory (then information is extracted
+                    # and all child directories are removed from search list).
+                    # Otherwise, we append the subdirs of this dir to search list (this
+                    # implements the recursive search)
+                    if dir.name not in ROLE_MANDATORY_DIRS:
+                        dir_contents.extend(
+                            [
+                                item
+                                for item in self.repo.directory_contents(
+                                    dir.path
+                                ).values()
+                                if item.type == "dir"
+                            ]
+                        )
+                    else:
+                        # get parent path
+                        role_path = os.path.split(dir.path)[0]
+                        # remove all role subdirs from the search list
+                        dir_contents = [
+                            item
+                            for item in dir_contents
+                            if not item.path.startswith(f"{role_path}/")
+                        ]
+
+                        # we want to get the git change history
```

Comment:
nit: This sounds a little like we want the whole git history, but we only want the timestamp of the last update (so the latest git change) ;-)

Reply:
updated